### ISSUE TITLE
Fix Request: AvoidRecompilingPattern missing case

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#180]
AvoidRecompilingPatterns doesn't detect if chained calls are use:
Example:

```
import java.util.regex.Pattern;

public class Foo {
    private static final Integer LIMIT = 11;
    private static final String SEP = "[.|,]";

    private boolean isAmountAllowed(String amount) {
        return LIMIT > Pattern.compile(SEP).matcher(amount).replaceAll("").length(); // bad
    }
    private boolean isAmountAllowed2() {
        String bla = Pattern.compile("[.|,]").matcher("5.000,00").replaceAll(""); //bad
        boolean b = (11 > bla.length());
        return b;
    }
}
```
